spec/src/main/asciidoc/integration.adoc

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright (c) 2023 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+== SDK integration
+
+Implementations SHALL provide integration by appropriately configuring compatible OpenTelemetry SDK for application runtime.
+Regardless of signal type being considered -- traces or metrics -- following requirements SHALL be met:
+
+[[sec:enabling-opentelemetry-support]]
+=== Enabling OpenTelemetry support
+By default, MicroProfile Telemetry is deactivated.
+
+In order to enable any of aspects of integration, the configuration `otel.sdk.disabled=false` MUST be specified in any of the configuration sources available via MicroProfile Config.
+
+[IMPORTANT]
+=====
+This is a deviation from the OpenTelemetry Specification that specifies this configuration property officially, where https://opentelemetry.io[OpenTelemetry] is activated by default!
+
+But in fact, it will be activated only by adding its dependency to the application or platform project.
+To be able to add MicroProfile Telemetry to MicroProfile implementations by default without side effects, this deviating behaviour has been defined here (see also <<sec:microprofile-telemetry-and-microprofile-opentracing>>).
+=====
+
+This property is read once when the application is starting.
+Any changes afterwards will not take effect unless the application is restarted.
+
+=== Configuration
+
+OpenTelemetry MUST be configured by MicroProfile Config following the semantics of configuration properties of link:https://github.com/open-telemetry/opentelemetry-java/tree/v{otel-java-version}/sdk-extensions/autoconfigure[OpenTelemetry SDK Autoconfigure extension].
```

Comment:
OpenTelemetry MUST be configured by MicroProfile Config based the semantics of configuration properties detailed in [https://github.com/open-telemetry/opentelemetry-java/tree/v{otel-java-version}/sdk-extensions/autoconfigure[OpenTelemetry](https://github.com/open-telemetry/opentelemetry-java/tree/v%7Botel-java-version%7D/sdk-extensions/autoconfigure%5BOpenTelemetry) SDK Autoconfigure {otel-java-version}].

Property names MAY be prefixed like prefix.otel.property. Prefixes MUST not colide with existing OpenTelemetry properties and the property names in the suffix part MUST match the ones defined in the OpenTelemetry SDK Autoconfigure.

Reply:
I'm not sure what this comment is trying to say.

This spec needs to specify which properties must be supported. For example, the user can set the MP config property `otel.traces.exporter=none` to disable the export of traces.

Implementations can always specify additional properties if they want to (for example allowing `acme.otel.traces.exporter` to be used and to override `otel.traces.exporter`). I don't think we need to have any text to explicitly allow that, or to restrict what properties an implementation may add.

However, I don't think we want to allow implementations to _require_ a prefix (i.e. to say that `otel.traces.exporter` alone does not work, the user must use `acme.otel.traces.exporter`).

Reply:
This spec should not say anything allowing or disallowing prefix but just state the property names it specs. Adding any over and above property names are implementation choices. That implementation should honor the properties specified in this specification as well as supporting the proprietary ones.

Reply:
CC @radcortez

Reply:
see above